src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs

```diff
@@ -34,7 +35,17 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
                 case StorageDatabase.SQLite:
                     var locationService = workspaceServices.GetService<IPersistentStorageLocationService>();
                     if (locationService != null)
-                        return new SQLitePersistentStorageService(locationService);
+                    {
+                        var experimentationService = workspaceServices.GetRequiredService<IExperimentationService>();
+                        if (experimentationService.IsExperimentEnabled(WellKnownExperimentNames.SQLiteInMemoryWriteCache))
```

Comment:
❗️ The experimentation service is disabled on some machines. This needs to go through a registry-backed option first, and only fall back to `IExperimentationService` if that is not set. Here is a good example:

https://github.com/dotnet/roslyn/blob/75fb12b5cb179a244830647729449e04eab47834/src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs#L61-L63

The `OOP64Bit` option can be set in cases where 64-bit OOP is desired but isn't enabled through the experimentation service.

Reply:
Sounds good!